### PR TITLE
Fix content item sync

### PR DESCRIPTION
**Pull request by ZhongYic00**
https://github.com/linuxdeepin/treeland/issues/183

### CODE REVIEW

File: src/server/qtquick/wsurfaceitem.cpp [hunk 1]
```diff
@@ -1221,7 +1213,7 @@ void WSurfaceItemPrivate::resizeSurfaceToItemSize(const QSize &itemSize, const Q
     }
 
     if (q->resizeSurface(itemSize)) {
-        contentContainer->setSize(contentContainer->size() + sizeDiff);
+        // contentContainer->setSize(contentContainer->size() + sizeDiff);
```
Comment (wineee) [inline src/server/qtquick/wsurfaceitem.cpp:1216]:
```suggestion
```

**Review by wineee — CHANGES_REQUESTED** [review]

File: src/server/qtquick/wsurfaceitem.cpp [hunk 2]
```diff
@@ -919,15 +919,7 @@ void WSurfaceItem::onSurfaceCommit()
...
     updateSurfaceState();
 
-    // Maybe the beforeRequestResizeSurfaceStateSeq is set by resizeSurfaceToItemSize,
-    // the resizeSurfaceToItemSize wants to resize the wl_surface to current size of WSurfaceitem,
-    // If change the WSurfaceItem's size at here, you will see the WSurfaceItem flash.
-    if (d->beforeRequestResizeSurfaceStateSeq < d->surface->handle()->handle()->current.seq) {
-        if (d->beforeRequestResizeSurfaceStateSeq != 0) {
-            Q_ASSERT(d->beforeRequestResizeSurfaceStateSeq == d->surface->handle()->handle()->current.seq - 1);
-            d->beforeRequestResizeSurfaceStateSeq = 0;
-        }
```
Comment (wineee) [inline src/server/qtquick/wsurfaceitem.cpp:929]:
```
  if (d->beforeRequestResizeSurfaceStateSeq <= d->surface->handle()->handle()->current.seq) {
        if (d->beforeRequestResizeSurfaceStateSeq != 0) {
            Q_ASSERT(d->beforeRequestResizeSurfaceStateSeq == d->surface->handle()->handle()->current.seq);
            d->beforeRequestResizeSurfaceStateSeq = 0;
        }
```

**Review by wineee — CHANGES_REQUESTED** [review]

File: src/server/qtquick/wsurfaceitem.cpp (hunk 2 above)
Comment (wineee) [reply src/server/qtquick/wsurfaceitem.cpp:929]: beforeRequestResizeSurfaceStateSeq is `pending.seq`, it should equal current.seq in next commit

File: src/server/qtquick/wsurfaceitem.cpp [hunk 3]
```diff
@@ -919,15 +919,7 @@ void WSurfaceItem::onSurfaceCommit()
...
-
+    if (d->surfaceState->contentSize!=d->contentContainer->size() || d->surfaceState->contentGeometry.topLeft()!=d->contentContainer->position()){
```
Comment (wineee) [inline src/server/qtquick/wsurfaceitem.cpp:922]: don't check this

File: src/server/qtquick/wsurfaceitem.cpp (hunk 2 above)
Comment (ZhongYic00) [reply src/server/qtquick/wsurfaceitem.cpp:929]: beforeRequestResizeSurfaceStateSeq is updated without waiting for ack configure, can skip many seq nums

**Comment by ZhongYic00** [comment]
hard to fix: #311

File: src/server/qtquick/wsurfaceitem.cpp [hunk 4]
```diff
@@ -1307,6 +1307,13 @@ qreal WSurfaceItemPrivate::getImplicitHeight() const
     return surfaceState->contentGeometry.height() + ps.height();
 }
 
+void WSurfaceItem::resizeBySurface(const QSize &newSize)
+{
+    W_D(WSurfaceItem);
+    Q_ASSERT( d->resizeMode == WSurfaceItem::SizeFromSurface);
```
Comment (wineee) [inline src/server/qtquick/wsurfaceitem.cpp:1313]:
```suggestion
    Q_ASSERT(d->resizeMode == WSurfaceItem::SizeFromSurface);
```

**Review by wineee — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]
Should sync example/tinywl-qtquick with a new PR.

**Comment by zccrs** [comment]
> Should sync example/tinywl-qtquick with a new PR.

@ZhongYic00

**Pull request merged**